src/AutoHashEqualsCached.jl

```diff
@@ -0,0 +1,240 @@
+# SPDX-License-Identifier: MIT
+
+module AutoHashEqualsCached
+
+export @auto_hash_equals_cached, @auto_hash_equals
+
+# `_show_default_auto_hash_equals_cached` is just like `Base._show_default(io, x)`,
+# except it ignores fields named `_cached_hash`.  This function is called in the
+# implementation of `T._show_default` for each type `T` annotated with
+# `@auto_hash_equals_cached`.  This is ultimately used in the implementation of
+# `Base.show`.  This specialization ensures that showing circular data structures does not
+# result in infinite recursion.
+function _show_default_auto_hash_equals_cached(io::IO, @nospecialize(x))
+    t = typeof(x)
+    show(io, Base.inferencebarrier(t)::DataType)
+    print(io, '(')
+    recur_io = IOContext(io, Pair{Symbol,Any}(:SHOWN_SET, x),
+                         Pair{Symbol,Any}(:typeinfo, Any))
+    if !Base.show_circular(io, x)
+        for i in 1:nfields(x)
+            f = fieldname(t, i)
+            if (f == :_cached_hash)
```

Comment:
nit: I think i've seen people prefer `===` for comparing literals like this. I'm not sure why, and it certainly doesn't make any material difference in the generated code, just a stylistic thing i guess?

Reply:
Fixed in #14.